### DISCUSSION TITLE
Netbox reports and custom fields

### DISCUSSION THREAD

**Discussion by nomad-cyanide** [netbox-community/netbox#7881] [General]
Hi

We have recently added custom fields to our Netbox installation, to be able to see status of our config-backup running each night.
I would like to create a report that shows all devices with their backup status, but creating a report correctly has proven somewhat of a pain.
I cannot find many usable examples and the documentation for this seems to just scratch the surface. 
Any thoughts on how I go forward from here?

Note I have tried various iterations of something like this, to just get the devices custom fields shown, without much luck:

```
from dcim.models import Device
from extras.models import CustomFieldValue

class BackupStatusReport(Report):
    description = "See status of last run config-backup for Juniper devices"
    
    def test_backup_status(self):
    
    	for device in Device.objects.all():
    		test = CustomFieldValue.objects.filter(obj_id=device.id)
    		if test:
        		self.log_info(
            		device,
            	"custom_fields on {} : {}".format(device.name, test)
        	)
```

Best regards from DK

**Comment by jeremystretch** [comment]
What version of NetBox are you running? CustomFieldValue was removed in v2.10 IIRC, and the newer solution is significantly easier to work with.

**Reply by nomad-cyanide** [reply]
We are on 2.11.4, so that would explain it. Thank you!
I will use my time and energy to get the upgrade to v3 done and then resume looking at this report.

**Reply by candlerb** [reply]
Custom fields work the same in both 2.11 and 3.0.  You access them using `device.custom_field_data`, or the shortcut alias `device.cf`.  It's just a dict of values, and stored in a JSON column.

Your code with `CustomFieldValue` would only be for Netbox 2.9 and earlier.
